### PR TITLE
Fix #28035 Refactor DLL imports interop classes

### LINKED ISSUE #28035: Properly factor DllImports into Common\src\Interop
We now have a bunch of libraries with their own DllImports, whereas the expected approach is that all DllImports in product src should be included from src\Common\src\Interop, appropriately factored by platform and library, and then those source files included into the relevant assemblies that need them.

- [ ] Microsoft.Diagnostics.Tracing.EventSource.Redist
- [ ] System.Data.SqlClient
- [x] System.Diagnostics.EventLog
- [ ] System.DirectoryServices
- [ ] System.DirectoryServices.AccountManagement
- [ ] System.DirectoryServices.Protocols
- [ ] System.Drawing.Common
- [x] System.IO.Compression
- [ ] System.IO.Compression.Brotli
- [ ] System.IO.Ports
- [x] System.Management
- [ ] System.Reflection.Metadata
- [x] System.Runtime.Caching
- [x] System.Security.Cryptography.Csp
- [x] System.Security.Cryptography.Pkcs
- [ ] System.Security.Cryptography.X509Certificates

This all needs to be refactored and dedup'd.

https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md

### PR DESCRIPTION
**Pull request by pedrobsaila**
Moved interop class to Common folder

#28035

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=48556) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=48556) before we can accept your contribution.<br/><br/>:x: pedrobsaila [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=48556)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=48556) it.</sub>

**Review by danmoseley — APPROVED** [review]
Thanks

**Comment by pedrobsaila** [comment]
You're welcome  👍

File: src/libraries/System.IO.Ports/src/System.IO.Ports.csproj [hunk 1]
```diff
@@ -120,8 +120,10 @@
   <ItemGroup Condition="'$(TargetsLinux)' == 'true' or '$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">
     <Compile Include="System\IO\Ports\SafeSerialDeviceHandle.Unix.cs" />
     <Compile Include="System\IO\Ports\SerialStream.Unix.cs" />
-    <Compile Include="Interop\Unix\Interop.Termios.cs" />
-    <Compile Include="Interop\Unix\Interop.Serial.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\Interop.Termios.cs"
```
Comment (jkotas) [inline src/libraries/System.IO.Ports/src/System.IO.Ports.csproj:123]: These files should be in `System.IO.Ports.Native` subdirectory - Interop\Unix\System.IO.Ports.Native\Interop.Termios.cs.

**Comment by danmoseley** [comment]
Appreciate it @pedrobsaila . Perhaps you'd like to pick up another up for grabs issue.

**Comment by danmoseley** [comment]
Actually, there's more to do in this issue , if you're interested.

**Pull request merged**